src/generation.rs

```diff
@@ -1,56 +1,74 @@
+use crate::cge_utils::Mutation;
+use crate::cmaes_utils::optimize_network;
+use crate::eant2::EANT2;
+use crate::utils::Individual;
+use crate::FitnessFunction;
+use cge::Network;
+use rand::{thread_rng, Rng};
+use rayon::prelude::*;
 use std::sync::Arc;
 
-use cge::{Network, Activation};
-use rand::thread_rng;
-use rand::distributions::{IndependentSample, Range};
+pub struct Generation<T: FitnessFunction + Clone> {
+  /// The individuals in the generation.
+  // TODO: reuse buffer, guess at capacity on fallback to new alloc.
+  // TODO: this will require something nice and clever to handle the fact that `T` is unknown.
+  pub(crate) individuals: Vec<Individual<T>>,
+}
 
-use crate::utils::Individual;
-use crate::cge_utils::Mutation;
-use crate::NNFitnessFunction;
-
-// Creates a generation of random, minimal neural networks
-pub fn initialize_generation<T>(population_size: usize,
-                                offspring_count: usize,
-                                inputs: usize,
-                                outputs: usize,
-                                activation: Activation,
-                                object: Arc<T>) -> Vec<Individual<T>>
-    where T: NNFitnessFunction + Clone
-{
+impl<T: FitnessFunction + Clone> Generation<T> {
+  /// Creates a generation of random, minimal neural networks.
+  pub fn initialize(options: &EANT2, object: Arc<T>) -> Generation<T> {
+    let individual_count = options.population * (1 + options.offspring);
 
+    // preallocate the exact amount of memory we know we will need (using an object pool in an ergonomic way is hard here because `T` is unknown).
+    let mut individuals = Vec::with_capacity(individual_count);
```

Comment:
`T` is unknown, but the only thing actually passed around is an `Arc<T>`, which can be safely cloned when creating an object pool. Still, I would recommend waiting until the API is completed just in case any of this changes.